utils.go

```diff
@@ -92,7 +93,7 @@ func DownloadFile(ctx context.Context, rawURL string) (outPath string, errRet er
 			return "", err
 		}
 		defer func() {
-			errRet = errors.Join(errRet, outfd.Close())
+			errRet = errors.Join(errRet, outfd.Close(), SyncFS(outPath))
```

Comment:
Similarly, to the degree this file needs to be removed, fsync the directory after calling `os.Remove`.